src/context/AppContext.tsx

```diff
@@ -37,7 +40,6 @@ export const AppContextProvider = ({ children }: { children: ReactNode }) => {
   try {
     initialWallet = getContext(ContextModule.Wallet);
   } catch (error) {
-    initialUser = null;
```

Comment:
If this is set to `null` here, user will never be `undefined` and thus we lose the distinction between a wallet without a registered user (`null`) and the state in which we're checking if there's a registered user (`undefined`)